cmd/proxy.go

```diff
@@ -134,18 +102,41 @@ func (s *socketMount) serve(ctx context.Context) error {
 		}
 		// handle the connection in a separate goroutine
 		go func() {
-			log.Printf("[%s] accepted connection from %s\n", s.inst, cConn.RemoteAddr())
-			sConn, err := s.dialer.Dial(ctx, s.inst)
+			pc.cmd.Printf("[%s] accepted connection from %s\n", s.inst, cConn.RemoteAddr())
+
+			// give a max of 30 seconds to connect to the instance
+			ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
```

Comment:
What's the relationship between this context and the on passed in above? Should we be building the new context here with the one passed in?